libs/dh/admin/feature-invite-user-modal/src/lib/dh-invite-user-modal.component.ts

```diff
@@ -117,65 +138,82 @@ export class DhInviteUserModalComponent implements AfterViewInit, OnDestroy {
 
   ngAfterViewInit(): void {
     this.inviteUserModal.open();
-    this.actorIdSubscription = this.userInfo.controls.actorId.valueChanges.subscribe((actorId) => {
+
+    this.emailExistsSubscription = this.userEmailExistsQuery.valueChanges.subscribe((x) => {
+      this.knownEmails = x.data?.knownEmails?.map((x) => x.toUpperCase()) ?? [];
+      this.isLoadingEmails = false;
+      this.changeDectorRef.detectChanges();
+    });
+
+    this.actorIdSubscription = this.baseInfo.controls.actorId.valueChanges.subscribe((actorId) => {
       actorId !== null
-        ? this.userInfo.controls.email.enable()
-        : this.userInfo.controls.email.disable();
+        ? this.baseInfo.controls.email.enable()
+        : this.baseInfo.controls.email.disable();
 
       if (actorId === null) {
         this.actorStore.resetOrganizationState();
         return;
       }
+
       this.assignableUserRolesStore.getAssignableUserRoles(actorId);
       this.actorStore.getActorOrganization(actorId);
     });
+
+    this.actorDomainSubscription = this.actorStore.organizationDomain$.subscribe((domain) => {
+      this.domain = domain;
+    });
+
+    this.actorEmailSubscription = this.baseInfo.controls.email.valueChanges.subscribe(
+      async (email) => {
+        this.inOrganizationMailDomain =
+          !!email &&
+          !!this.domain &&
+          this.baseInfo.controls.email.valid &&
+          email.toUpperCase().endsWith(this.domain.toUpperCase());
+
+        this.emailExists =
+          !!email &&
+          this.baseInfo.controls.email.valid &&
+          this.knownEmails.includes(email.toUpperCase());
+
+        this.changeDectorRef.detectChanges();
+      }
+    );
+
+    this.actorsSubscription = this.actors$.subscribe((actors) => {
+      if (actors.length === 1) {
+        this.baseInfo.controls.actorId.setValue(actors[0].value);
+      }
+    });
   }
 
   ngOnDestroy(): void {
+    this.emailExistsSubscription?.unsubscribe();
     this.actorIdSubscription?.unsubscribe();
+    this.actorDomainSubscription?.unsubscribe();
+    this.actorEmailSubscription?.unsubscribe();
+    this.actorsSubscription?.unsubscribe();
   }
 
   inviteUser() {
-    if (this.userInfo.valid === false || this.userRoles.valid === false) {
+    if (!this.isBaseInfoValid() || !this.isNewUserInfoValid() || !this.isRolesInfoValid()) {
       return;
     }
 
-    const { firstname, lastname, email, phoneNumber, actorId } = this.userInfo.controls;
-
-    const emailWithDomain = `${email.value}@${this.domain}`;
+    const { firstname, lastname, phoneNumber } = this.userInfo.controls;
+    const { email, actorId } = this.baseInfo.controls;
 
     this.inviteUserStore.inviteUser({
       invitation: {
-        firstName: firstname.value ?? '',
-        lastName: lastname.value ?? '',
-        email: emailWithDomain,
-        phoneNumber: phoneNumber.value ?? '',
+        firstName: firstname.value ? firstname.value : 'J',
```

Comment:
Suggestion: Make a guard clause with an early return for the required values instead of strange default values.

Reply:
Will be fixed later. API will be changed.